src/common/crc32c_s390x.c

```diff
@@ -0,0 +1,580 @@
+#include <sys/types.h>
+#include <endian.h>
+#include "crc32c_s390x.h"
+
+#define VX_MIN_LEN		64
+#define VX_ALIGNMENT		16L
+#define VX_ALIGN_MASK		(VX_ALIGNMENT - 1)
+
+/* CRC-32C slicing-by-8 constants, for use on big-endian systems */
+static const unsigned int __attribute__((aligned(128))) crc32ctable_le[8][256] = {
+    {
+    0x00000000, 0x03836bf2, 0xf7703be1, 0xf4f35013,
+    0x1f979ac7, 0x1c14f135, 0xe8e7a126, 0xeb64cad4,
+    0xcf58d98a, 0xccdbb278, 0x3828e26b, 0x3bab8999,
+    0xd0cf434d, 0xd34c28bf, 0x27bf78ac, 0x243c135e,
+    0x6fc75e10, 0x6c4435e2, 0x98b765f1, 0x9b340e03,
+    0x7050c4d7, 0x73d3af25, 0x8720ff36, 0x84a394c4,
```

Comment:
where did this table come from? do we need to include a copyright/license attribution?

Reply:
> where did this table come from? do we need to include a copyright/license attribution?

It is taken from the crc code:
https://github.com/linux-on-ibm-z/crc32-s390x/blob/master/slicing-consts.h
I add the IBM copyright.

Reply:
> I add the IBM copyright.

Did this happen?

Reply:
Also, since https://github.com/linux-on-ibm-z/crc32-s390x/blob/master/LICENSE.TXT allows redistribution under either GPL 2 and later or Apache 2.0 and Ceph is generally licensed under LGPL, an exception record needs to be added to https://github.com/ceph/ceph/blob/main/COPYING.